plugins/balancedLevelProvider.go

```diff
@@ -30,6 +30,12 @@ type balancedLevelProvider struct {
 
 	// precomputed before construction
 	randGen *rand.Rand
+
+	//keeps the starting balances from the previous run to check if anything has changed
+	lastMaxAssetBase  float64
```

Comment:
these three variables can be in a new section of: `// uninitialized`.
this section will come after `randGen`